src/act/node/data_type/variant.rs

```diff
@@ -0,0 +1,122 @@
+use proc_macro2::TokenStream;
+use quote::{quote, ToTokens};
+use std::collections::HashMap;
+
+use super::{traits::ToTypeAnnotation, DataType};
+use crate::{
+    act::{node::traits::HasMembers, proclamation::Proclaim},
+    keyword,
+    traits::ToIdent,
+};
+
+#[derive(Clone, Debug)]
+pub struct Variant {
+    pub name: Option<String>,
+    pub members: Vec<Member>,
+}
+
+#[derive(Clone, Debug)]
+pub struct Member {
```

Comment:
Move this to live with HasMember trait and use it with Record as well

Reply:
I'm not so sure about this anymore. They both have wildly different to_token_stream implementations, so I think we actually get little by moving these out compared to the overhead that we would incur. Maybe we can still do it, but it will require more careful thought. I'll come back to this one.

Reply:
I've decided against it. They are different Members, and the impl for them proves that. they might have the same data but they have very different impls, I understand that classes and objects shouldn't be the standard that we go by, but think about making a class for a person and a bottle of wine. They might be defined by only two fields each, name and age, but all of the other details of that class are going to be very different. Is the wine and person a little ridiculous? Maybe, but it illustrates the point